contrib/database/sql/conn.go

```diff
@@ -199,10 +175,10 @@ func WithSpanTags(ctx context.Context, tags map[string]string) context.Context {
 	return context.WithValue(ctx, spanTagsKey, tags)
 }
 
-// withSQLCommentsInjected will return the query with sql comments injected according to the comment injection mode along
+// injectComments returns the query with sql comments injected according to the comment injection mode along
 // with a span id injected into sql comments. If a span ID is returned, the caller should make sure to use it when creating
 // the span following the traced database call.
-func (tp *traceParams) withSQLCommentsInjected(ctx context.Context, query string, discardDynamicTags bool) (cquery string, injectedSpanID uint64) {
+func (tp *traceParams) injectComments(ctx context.Context, query string, discardTracingTags bool) (cquery string, spanID uint64) {
```

Comment:
Since this method uses nothing from `tp`, it doesn't have to be a method, it can be simply a function.
```suggestion
func injectComments(ctx context.Context, query string, discardTracingTags bool) (cquery string, spanID uint64) {
```